File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/isolationpayment/model/TokenUpdateRequest.java
package uk.nhs.nhsx.isolationpayment.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import uk.nhs.nhsx.virology.IpcTokenId;

import java.time.Instant;

public class TokenUpdateRequest {
    public final IpcTokenId ipcToken;
    public final Instant riskyEncounterDate;
    public final Instant isolationPeriodEndDate;

    @JsonCreator
    public TokenUpdateRequest(IpcTokenId ipcToken, Instant riskyEncounterDate, Instant isolationPeriodEndDate) {
        this.ipcToken = ipcToken;
        this.riskyEncounterDate = riskyEncounterDate;
        this.isolationPeriodEndDate = isolationPeriodEndDate;
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/isolationpayment/model/TokenUpdateRequest.kt
package uk.nhs.nhsx.isolationpayment.model

import uk.nhs.nhsx.domain.IpcTokenId
import java.time.Instant

data class TokenUpdateRequest(
    val ipcToken: IpcTokenId,
    val riskyEncounterDate: Instant,
    val isolationPeriodEndDate: Instant
)
